### PR TITLE
[#912] Harden access control on case-details/documents

### PR DESCRIPTION
**Pull request by Bartvaderkin**
Continues from PR #327 & taiga 890

### CODE REVIEW

**Review by vaszig — COMMENTED** [review]
This has been reviewed in #327  except for the addition for the user's roles. So, it seems ok to me and can be merged after the changes-update of the mentioned PR.

**Comment by Bartvaderkin** [comment]
I'll rebase this once the other changes land and remark for review as Anna hasn't looked at this yet.

**Comment by Bartvaderkin** [comment]
I've rebased this on the PR for #890 and marked this as ready for review but note the review only covers the commit marked with [#912] (the others marked [#890] are part of the other PR)

File: src/open_inwoner/accounts/views/cases.py
Comment (annashamray) [inline src/open_inwoner/accounts/views/cases.py:160]: TBH checking permissions in `get_context_data` seems odd and scattering permission checks between `test_func` and  `get_context_data` makes it hart to understand the authorization logic. 
I'd suggest putting all permissions in one place (`test_func` or `get` or `dispatch`)

Since we already use mixins for permissions and this code is also used in `CaseDocumentDownloadView` it can also be put in the separate mixin

File: src/open_inwoner/accounts/views/cases.py
Comment (annashamray) [inline src/open_inwoner/accounts/views/cases.py:262]: I'd also suggest to put `test_func` and permission checks in `get` in one place

**Comment by Bartvaderkin** [comment]
@annashamray I made an attempt to your suggestion to centralize the access checks: I dumped the regular mixins and turned them into a dispatch()-override because of the order in which the mixins super().dispatch() methods would execute (and we want auth/bsn check before we grab data).

Feel free to merge/append/refactor if this holds up the case confidentiality tickets

**Review by annashamray — APPROVED** [review]
Looks nice!

**Review by alextreme — APPROVED** [review]

**Pull request merged**
